internal/backend/store/store.go

```
@@ -65,6 +80,35 @@ func (s *store) Mutate(ctx context.Context, pid sdktypes.ProjectID, key, op stri
 		}
 
 		if r.write {
+			if s.cfg.MaxValueSize != 0 && next.ProtoSize() > s.cfg.MaxValueSize {
+				return sdkerrors.NewInvalidArgumentError("value size %d exceeds maximum allowed size %d for a single value", next.ProtoSize(), s.cfg.MaxValueSize)
+			}
+
+			if maxCount := int64(s.cfg.MaxStoreKeysPerProject); maxCount > 0 {
```

why would this ever be zero ? 
and if it can be zero, why do we check it here ? 
you can't read / write anything if the max is zero, better fail fast at the top of the function

---

if it's 0 - no limit. see comment in config.